File: scala-play/app/services/v2/EchoService.scala
package services.v2

import javax.inject._
import scala.concurrent._
import models.v2._

@Singleton
class EchoService @Inject()()(implicit ec: ExecutionContext) extends IEchoService {
  import IEchoService._
  override def echoBody(body: Message): Future[EchoBodyResponse] = Future {
    EchoBodyResponse.Ok(Message(body.boolField, body.stringField))
  }
}

File: ts-koa/src/v2/echo_service.ts
import * as service from '../spec/v2/echo_service'
import * as models from '../spec/v2/models'

export let echoService = (): service.EchoService => {
    let echoBody = async (params: service.EchoBodyParams): Promise<models.Message> => {
        return {bool_field: params.body.bool_field, string_field: params.body.string_field}
    }
    
    return {echoBody}
}